settings-manager.js

```diff
@@ -0,0 +1,42 @@
+class SettingsManager extends EventTarget {
+  constructor() {
+    super();
+
+    this.addEventListener('gfxSettingsChanged', e => {
+      const settings = e.data;
+      this.#setSettings('GfxSettings', settings);
```

Comment:
Does it make sense to separate these events at all?

Is there ever a case, besides the settings panel itself, where we are generally interested in a page of settings as opposed to a setting itself? What would be the purpose of such a notification/categorizing it this way for users of this API, other than confusion?

IMO it makes more sense to emit that a specific setting value changed.

Reply:
i'll change it to a single event.

I also agree that a single value makes more sense.  It's currently accepting the full set of perameters because of how the saveSettings functions are written in the settings tabs, But I'll adjust accordingly